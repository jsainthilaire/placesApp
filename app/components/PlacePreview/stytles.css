.container {
    background-color: #f5f3f3;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 6px;
    padding: 10px;
    border-radius: 10px;
}

.placeImage {
    width: 300px;
}