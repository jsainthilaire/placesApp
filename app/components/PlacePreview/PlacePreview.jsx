import React from 'react'
import PropTypes from 'prop-types'
import { placeImage, container } from './stytles.css'

const PlacePreview = ({
  name,
  description,
  imageURL,
  saveToUser,
  isUserPlaces,
}) => (
  <article className={container}>
    <h2>{name}</h2>
    <p>{description}</p>
    <img src={imageURL} className={placeImage} />
    {!isUserPlaces && <button onClick={saveToUser}>Save</button>}
  </article>
)

PlacePreview.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  imageURL: PropTypes.string.isRequired,
  saveToUser: PropTypes.func.isRequired,
  isUserPlaces: PropTypes.bool.isRequired,
}

export default PlacePreview
