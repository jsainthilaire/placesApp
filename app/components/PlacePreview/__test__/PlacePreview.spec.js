import React from 'react'
import { shallow } from 'enzyme'
import noop from 'lodash/noop'
import toJson from 'enzyme-to-json'
import PlacePreview from '../PlacePreview'

const minProps = {
  name: '',
  description: '',
  imageURL: '',
  saveToUser: noop,
  isUserPlaces: false,
}

test('PlacePreview should render correctly', () => {
  const wrapper = shallow(<PlacePreview {...minProps} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

test('PlacePreview should not render save button when the page is my places', () => {
  const wrapper = shallow(<PlacePreview {...minProps} isUserPlaces />)
  expect(wrapper.find('button')).toHaveLength(0)
  expect(toJson(wrapper)).toMatchSnapshot()
})

test('PlacePreview should render save button when the page is my places', () => {
  const wrapper = shallow(<PlacePreview {...minProps} isUserPlaces={false} />)
  expect(wrapper.find('button')).toHaveLength(1)
  expect(toJson(wrapper)).toMatchSnapshot()
})

test('PlacePreview should call saveToUser when the button is clicked', () => {
  const saveToUser = jest.fn()
  const wrapper = shallow(<PlacePreview {...minProps} saveToUser={saveToUser} />)
  wrapper.find('button').simulate('click')
  expect(saveToUser).toBeCalled()
})
