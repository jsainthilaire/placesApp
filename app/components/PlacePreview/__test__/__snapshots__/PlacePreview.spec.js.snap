// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacePreview should not render save button when the page is my places 1`] = `
<article
  className="container"
>
  <h2 />
  <p />
  <img
    className="placeImage"
    src=""
  />
</article>
`;

exports[`PlacePreview should render correctly 1`] = `
<article
  className="container"
>
  <h2 />
  <p />
  <img
    className="placeImage"
    src=""
  />
  <button
    onClick={[Function]}
  >
    Save
  </button>
</article>
`;

exports[`PlacePreview should render save button when the page is my places 1`] = `
<article
  className="container"
>
  <h2 />
  <p />
  <img
    className="placeImage"
    src=""
  />
  <button
    onClick={[Function]}
  >
    Save
  </button>
</article>
`;
